main.tf

```diff
@@ -130,9 +166,81 @@ resource "aws_lb_listener" "frontend_https" {
+      target_group_arn = contains([null, "", "forward"], lookup(default_action.value, "action_type", "")) ? aws_lb_target_group.main[lookup(default_action.value, "target_group_index", count.index)].id : null
+
+      dynamic "redirect" {
+        for_each = length(keys(lookup(default_action.value, "redirect", {}))) == 0 ? [] : [lookup(default_action.value, "redirect", {})]
+
+        content {
+          path        = lookup(redirect.value, "path", null)
+          host        = lookup(redirect.value, "host", null)
+          port        = lookup(redirect.value, "port", null)
+          protocol    = lookup(redirect.value, "protocol", null)
+          query       = lookup(redirect.value, "query", null)
+          status_code = lookup(redirect.value, "status_code", null)
+        }
+      }
+
+      dynamic "fixed_response" {
+        for_each = length(keys(lookup(default_action.value, "fixed_response", {}))) == 0 ? [] : [lookup(default_action.value, "fixed_response", {})]
+
+        content {
+          content_type = lookup(fixed_response.value, "content_type", null)
+          message_body = lookup(fixed_response.value, "message_body", null)
+          status_code  = lookup(fixed_response.value, "status_code", null)
+        }
+      }
+
+      # Authentication actions only available with HTTPS listeners
+      dynamic "authenticate_cognito" {
+        for_each = length(keys(lookup(default_action.value, "authenticate_cognito", {}))) == 0 ? [] : [lookup(default_action.value, "authenticate_cognito", {})]
+
+        content {
+          # Max 10 extra params
+          authentication_request_extra_params = lookup(authenticate_cognito.value, "authentication_request_extra_params", null)
+          on_unauthenticated_request          = lookup(authenticate_cognito.value, "on_authenticated_request", null)
+          scope                               = lookup(authenticate_cognito.value, "scope", null)
+          session_cookie_name                 = lookup(authenticate_cognito.value, "session_cookie_name", null)
+          session_timeout                     = lookup(authenticate_cognito.value, "session_timeout", null)
+          user_pool_arn                       = lookup(authenticate_cognito.value, "user_pool_arn", null)
+          user_pool_client_id                 = lookup(authenticate_cognito.value, "user_pool_client_id", null)
+          user_pool_domain                    = lookup(authenticate_cognito.value, "user_pool_domain", null)
+        }
+      }
+
+      dynamic "authenticate_oidc" {
+        for_each = length(keys(lookup(default_action.value, "authenticate_oidc", {}))) == 0 ? [] : [lookup(default_action.value, "authenticate_oidc", {})]
+
+        content {
+          # Max 10 extra params
+          authentication_request_extra_params = lookup(authenticate_oidc.value, "authentication_request_extra_params", null)
+          authorization_endpoint              = lookup(authenticate_oidc.value, "authorization_endpoint", null)
```

Comment:
Some of the arguments are required (for eg, `authorization_endpoint`), so we can't use `null` for them.

It is better to write `authorization_endpoint = authenticate_oidc.value["authorization_endpoint"]`, and let it fail when user don't specify it during plan.

Please review and update all arguments of actions blocks as specified here: https://www.terraform.io/docs/providers/aws/r/lb_listener.html

Reply:
Haven't really though about that. Just updated it.